backend/dataall/cdkproxy/stacks/policies/glue.py

```diff
@@ -82,36 +144,52 @@ def get_statements(self):
                 ],
                 conditions={
                     'StringEquals': {f'aws:RequestTag/{self.tag_key}': [self.tag_value]}
-                },
+                }
             ),
             iam.PolicyStatement(
-                actions=[
-                    'glue:DeleteDevEndpoint',
-                    'glue:DeleteCrawler',
-                    'glue:DeleteJob',
-                    'glue:DeleteTrigger',
-                    'glue:StartCrawler',
-                    'glue:StartTrigger',
-                    'glue:StopCrawler',
-                    'glue:StopTrigger',
-                    'glue:UpdateCrawler',
-                    'glue:UpdateDevEndpoint',
-                    'glue:UpdateJob',
-                    'glue:UpdateTrigger',
+                #sid="GlueManageGlueResources",
+                effect=iam.Effect.ALLOW,
+                not_actions=[
+                    'glue:CreateDevEndpoint',
+                    'glue:CreateTrigger',
+                    'glue:CreateJob',
+                    'glue:CreateCrawler',
                 ],
                 resources=[
-                    f'arn:aws:glue:{self.region}:{self.account}:crawler/{self.resource_prefix}*',
-                    f'arn:aws:glue:{self.region}:{self.account}:job/{self.resource_prefix}*',
                     f'arn:aws:glue:{self.region}:{self.account}:devEndpoint/{self.resource_prefix}*',
-                    f'arn:aws:glue:{self.region}:{self.account}:catalog',
                     f'arn:aws:glue:{self.region}:{self.account}:trigger/{self.resource_prefix}*',
-                    f'arn:aws:glue:{self.region}:{self.account}:table/{self.resource_prefix}*/*',
+                    f'arn:aws:glue:{self.region}:{self.account}:job/{self.resource_prefix}*',
+                    f'arn:aws:glue:{self.region}:{self.account}:crawler/{self.resource_prefix}*'
                 ],
                 conditions={
                     'StringEquals': {
                         f'aws:resourceTag/{self.tag_key}': [self.tag_value]
                     }
                 },
             ),
+            iam.PolicyStatement(
+                #sid="SupportGluePermissions",
+                effect=iam.Effect.ALLOW,
+                actions=[
+                    'glue:*Classifier',
+                    'glue:CreateScript',
+                    'glue:CreateSecurityConfiguration',
+                    'glue:DeleteSecurityConfiguration',
```

Comment:
can we limit the scope for the create/delete security configuration operations?

Reply:
It is not possible to scope down this action ([docs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsglue.html#awsglue-resources-for-iam-policies)). Our alternatives are:
1. Allow only creation, so that we do not delete any other-team's security configuration
2. Do not allow anything and let the AWS Account admins manage the security configurations, we can then use them in our Glue Jobs....

What do you think?

Reply:
if Creation is mandatory, then we should consider removing the Delete permission. if we can support using configurations created outside of dataall, then that would be safest.